javaslang/src/test/java/javaslang/collection/AbstractSeqTest.java

```diff
@@ -1217,6 +1217,27 @@ public void shouldCreateReverseIteratorOfNonEmpty() {
         assertThat(ofAll(of("a", "b", "c")).reverseIterator().toList()).isEqualTo(Iterator.of("c", "b", "a").toList());
     }
 
+    // -- shuffle
+
+    @Test
+    public void shouldShuffleEmpty() {
+        assertThat(empty().shuffle().isEmpty());
+    }
+
+    @Test
+    public void shouldShuffleHaveSameLength() {
+        assertThat(of(1, 2, 3).shuffle().size()).isEqualTo(of(1, 2, 3).size());
+    }
+
+    @Test
+    public void shouldShuffleHaveSameElements() {
+        Seq<Integer> shuffled = of(1, 2, 3).shuffle();
+        assertThat(shuffled.indexOf(1)).isNotEqualTo(-1);
```

Comment:
I did consider it, but I thought that `indexOf` was more primitive and therefore would limit the scope of what's being tested a little more than the use of a higher level function would.

It's up to you whether it gets rewritten or not, I'll do it if you think it needs doing.